### PR TITLE
[fli-iam#2192] New properties API endpoints

### PR DESCRIPTION
**Pull request by youennmerel**
# Changes
Add 2 endpoints in dataset MS to query dataset properties
 - GET /properties/dataset/{datasetId}
   - check user CAN_SEE_ALL right on given dataset
 - GET /properties/processing/{processingId}
   - keep properties for which user has CAN_SEE_ALL right on dataset

# How to test
- Check datasets swagger doc
- Check APIs with cURL (copy a query example "as cURL" in browser console and adapt it)
- Check different user rights

### CODE REVIEW

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/property/controller/DatasetPropertyApiController.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        return new ResponseEntity<>(
+                mapper.datasetPropertiesToDatasetPropertyDTOs(this.propertyService.getByDatasetId(datasetId)),
+                HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<List<DatasetPropertyDTO>> getPropertiesByProcessingId(Long processingId) throws EntityNotFoundException {
+        List<DatasetProperty> filteredProperties = new ArrayList<>();
+
+        for(DatasetProperty property : this.propertyService.getByDatasetProcessingId(processingId)){
```
[the remark this answers is not in the document]
Comment (jcomedouteau) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/property/controller/DatasetPropertyApiController.java:44]: Okay, this works, it returns an empty array :+1:

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/property/controller/DatasetPropertyApiController.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @Autowired
+    private DatasetSecurityService securityService;
+
+    @Autowired
+    private DatasetPropertyMapper mapper;
+
+
+    @Override
+    public ResponseEntity<List<DatasetPropertyDTO>> getPropertiesByDatasetId(Long datasetId) {
+        return new ResponseEntity<>(
```
[the remark this answers is not in the document]
Comment (jcomedouteau) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/property/controller/DatasetPropertyApiController.java:35]: It works too, returns an empty array. :+1:

**Review by jcomedouteau — APPROVED** [review]
Just the little correction og the doc, otherwise, all good for me !

**Comment by youennmerel** [comment]
> Just the little correction og the doc, otherwise, all good for me !

Hi @jcomedouteau,

Thanks for your review, OK now !

**Pull request merged**